### ISSUE TITLE
Connect Odoo pod to Postgres cluster (Patroni)

### ISSUE THREAD

**Issue by gridworkz** [bitnami/bitnami-docker-odoo#59]
Hi Javier:

Sorry, I was a little vague. I use the following parameters to the Bitnami Odoo Helm chart:

odooUsername=admin
odooPassword=gppRp86SDK
image.tag=12.0
ingress.hosts[0].name=gridworkz.com
ingress.hosts[0].tls=true
ingress.hosts[0].certManager=true
ingress.hosts[0].tlsSecret=gridworkz-com-tls-prod
persistence.storageClass=gridworkz-nfs-sc1
postgresql.postgresqlPassword=tea **(the superuser postgres pw on the Patroni cluster)** 
postgresql.persistence.storageClass=openebs-cstor-sparse

I need to set db_host to: patroni.patroni.svc.cluster.local in odoo-server.conf and point to an external postgres cluster running Patroni. If I manually change the odoo-server.conf file after installing the Helm chart I can't reach the cluster and I get that error:

nami    ERROR Unable to start com.bitnami.odoo: Traceback (most recent call last):
  File "odoo-server", line 9, in <module>
    odoo.cli.main()
  File "/opt/bitnami/odoo/odoo/cli/command.py", line 60, in main
    o.run(args)
  File "/opt/bitnami/odoo/odoo/cli/server.py", line 175, in run
    main(args)
  File "/opt/bitnami/odoo/odoo/cli/server.py", line 140, in main
    odoo.service.db._create_empty_database(db_name)
  File "/opt/bitnami/odoo/odoo/service/db.py", line 95, in _create_empty_database
    with closing(db.cursor()) as cr:
  File "/opt/bitnami/odoo/odoo/sql_db.py", line 637, in cursor
    return Cursor(self.__pool, self.dbname, self.dsn, serialized=serialized)
  File "/opt/bitnami/odoo/odoo/sql_db.py", line 178, in __init__
    self._cnx = pool.borrow(dsn)
  File "/opt/bitnami/odoo/odoo/sql_db.py", line 520, in _locked
    return fun(self, *args, **kwargs)
  File "/opt/bitnami/odoo/odoo/sql_db.py", line 588, in borrow
    **connection_info)
  File "/opt/bitnami/odoo/venv/lib/python3.6/site-packages/psycopg2/__init__.py", line 130, in connect
    conn = _connect(dsn, connection_factory=connection_factory, **kwasync)
psycopg2.OperationalError: could not create SSL context: library has no ciphers
FATAL:  pg_hba.conf rejects connection for host "10.42.4.43", user "bn_odoo", database "postgres", SSL off

I've tried various combinations of db_user / db_password and I also tried altering the environment variables controlling the Odoo pod but still no luck. It's as though Odoo (or at least psycopg2) can't see the pg_hba.conf file in the cluster. I can shell into the Odoo pod and psql to the cluster no problem but when I put these same creds into odoo-server.conf I get the above error.

btw: There is a bn_odoo user on the cluster and I've even installed the bitnami_odoo db there as well. I can connect to it with: psql -U bn_odoo -h patroni.patroni.svc.cluster.local -d bitnami_odoo from the command line no problem (from the Odoo pod).

Basically, I need to know if I can set: db_user / db_password / db_host / db_name in the values.yaml file then launch stable/odoo with Helm from the command-line.

Cheers,
Dave

**Comment by juan131** [comment]
Hi @gridworkz 

I'm afraid there's no way to run the Odoo chart with an external database at this very moment. It'll require several changes both in the chart and the docker image.

This is sth supported on other charts (e.g. Redmine). Those charts define conditions that allow the user to choose wether to use a external database or not, as you can see below:

https://github.com/helm/charts/blob/master/stable/redmine/requirements.yaml#L5
https://github.com/helm/charts/blob/master/stable/redmine/requirements.yaml#L9

As you can see in the [README.md](https://github.com/helm/charts/blob/master/stable/redmine/README.md), there are parameters that allow selecting an external database:


Parameter | Description | Default
-- | -- | --
externalDatabase.host | Host of the external database | localhost
externalDatabase.user | External db admin user | root
externalDatabase.password | Password for the admin user | ""
externalDatabase.port | Database port number | 330

I guess you're looking for a solutions such as this we implemented in the Redmine chart, aren't you?

**Comment by gridworkz** [comment]
Hi Juan:

Yes, correct, the Redmine functionality is what I'd be looking for. Is that
a big effort?

Cheers,
Dave

**Comment by juan131** [comment]
Hi @gridworkz 

> Is that a big effort?

As I mentioned in the previous comment, supporting external database will require several changes both in the chart and the docker image.

I added your request to our internal backlog and it'll be evaluated and planned to be implemented in the future. 

We'll keep you updated and ping you in this thread once we add support for external databases.

**Comment by gridworkz** [comment]
Hi Juan:

I appreciate that...

Cheers,
Dave

**Comment by thanhtai9606** [comment]
It just delete user account **bn_odoo**  in pgadmin, it work fine
